src/test/java/org/rxbooter/injector/core/KeyTest.java

```diff
@@ -78,6 +78,7 @@ public void keyFollowsEqualsContract() throws Exception {
         assertThat(key1).isEqualTo(key1);
         assertThat(key2).isEqualTo(key2);
 
+        //Key compared to string ??
         assertThat(key1.equals("")).isFalse();
```

Comment:
This class has some specifics - boolean supplier is not taken into account for equals and hashCode.